### PR TITLE
Add support for Unicorn HAT HD

### PR DESCRIPTION
**Pull request by rm-hull**
https://shop.pimoroni.com/products/unicorn-hat-hd

- [x] Working implementation
- [x] Documentation
- [x] Regression tests

### CODE REVIEW

File: doc/intro.rst [hunk 1]
```diff
@@ -1,9 +1,10 @@
 Introduction
 ------------
 Python library interfacing LED matrix displays with the MAX7219 driver (using
-SPI) and WS2812 & APA102 NeoPixels on the Raspberry Pi and other linux-based
-single board computers - it provides a Pillow-compatible drawing canvas, and
-other functionality to support:
+SPI) and WS2812 & APA102 NeoPixels (inc Pimoroni Unicorn pHat/Hat and Unicorn
+Hat HD) on the Raspberry Pi and other linux-based single board computers - it
```
Comment (thijstriemstra) [inline doc/intro.rst]: 'Linux'

File: doc/python-usage.rst [hunk 2]
```diff
@@ -354,6 +354,31 @@ a translation mapping is required, as follows:
 
 This should animate a green dot moving left-to-right down each line.
 
+Pimoroni Unicorn HAT HD
+"""""""""""""""""""""""
+Pimoroni sells the `Unicorn HAT HD <https://shop.pimoroni.com/products/unicorn-hat-hd>`_, 
+comprising 256 high-intensity RGB LEDs a 16x16 arrangement. The pixels driven by ARM STM32F
```
Comment (thijstriemstra) [inline doc/python-usage.rst]: 'a 16x16'

File: doc/python-usage.rst [hunk 3]
```diff
@@ -354,6 +354,31 @@ a translation mapping is required, as follows:
...
+making the display appear as an SPI device:
+
+.. code:: python
+
+    import time
```
Comment (thijstriemstra) [inline doc/python-usage.rst]: would be better to use monotonic time here but, since it's not in Py27, it would overcomplicate the example

Comment (rm-hull) [reply doc/python-usage.rst]: \*nod\*

File: tests/test_unicornhathd.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+def test_display():
+    device = unicornhathd(serial)
+    serial.reset_mock()
+
+    with canvas(device) as draw:
+        draw.rectangle(device.bounding_box, outline="white")
+
+    serial.data.assert_called_once_with(
+        [0x72] + [112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112, 112,
```
Comment (thijstriemstra) [inline tests/test_unicornhathd.py]: this should be moved to json i guess? any other tests use this long array-type data? can be handled in a different ticket though.

**Pull request merged**